MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm

```diff
@@ -108,13 +108,13 @@
 		mslVerEnum = MTLLanguageVersion1_1;
 	}
 
-	if ( !mslVerEnum ) {
-		errMsg = [NSString stringWithFormat: @"%d.%d.%d is not a valid MSL version number on this device",
-				  mslVersionMajor, mslVersionMinor, mslVersionPoint].UTF8String;
-		return false;
-	}
-
 	@autoreleasepool {
+		if ( !mslVerEnum ) {
+			errMsg = [NSString stringWithFormat: @"%d.%d.%d is not a valid MSL version number on this device",
+					  mslVersionMajor, mslVersionMinor, mslVersionPoint].UTF8String;
+			return false;
+		}
+
```

Comment:
This will cause the error message string to be released and destroyed immediately before the function returns, because the autorelease pool will be torn down at that point. (ARC is off in MoltenVK, remember?) You may need to add a `-retain` call here... which would, in turn, entail adding `-release` calls to this function's callers.

Reply:
Wait, will it? I was operating on the assumption that the `-[NSString string...]` family of methods return autoreleased, which is part of why they need to be called in an autorelease pool in the first place, but I might be mistaken.

Reply:
Wait, it doesn't even matter, because you get it as a `UTF8String` right away and then copy it into the `errMsg`. Sorry about that.

Reply:
Yes, the `-[NSString string...]` family returns an autoreleased `NSString` object, so it will be destroyed by the autorelease pool.

However, the subtlety here is that  `errMsg` is a `std::string`, and the assignment will copy the internal `UTF8String` from the `NSString` object into the `std::string`.